src/Uno.UI/UI/Xaml/UIElement.Pointers.wasm.cs

```diff
@@ -249,7 +264,16 @@ public static void OnNativePointerEvent()
 		const int cancel = (int)NativePointerEvent.pointercancel;
 		const int exitOrUp = (int)(NativePointerEvent.pointerout | NativePointerEvent.pointerup);
 
-		var pointerId = (uint)args.pointerId;
+		var pointerId = TransformPointerId((uint)args.pointerId);
+		if ((NativePointerEvent)args.Event == NativePointerEvent.pointerdown)
+		{
+			s_activePointers.Add(pointerId);
+		}
+		else if ((NativePointerEvent)args.Event == NativePointerEvent.pointerup)
+		{
+			s_activePointers.Remove(pointerId);
```

Comment:
Yeah that's the good way to go 👍

The issue here is that you will remove the entry on the first "pointerup" before the event bubbles. But events are bubbling in native on wasm (unless they are flagged as handled) so this method is going to be invoked also for all parent elements (as soon as they have pointer event handler registered) which might drive to have inconsistent ID.

Fortunately we now a unique root element where we can handle such kind of requirement, the `RootVisual`. So you should move the cleanup logic here: https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Internal/RootVisual.cs#L143. You will also have to cleanup on `PointerCancelled`.

Also, I would do that only for touch pointers if possible, it would avoid useless dictionary lookup.

Reply:
> Also, I would do that only for touch pointers if possible

What about "Pen" pointers? I don't have a stylus device to be able to test :(

Reply:
Well, I could test but, yeah it's probably safer to it for all pointer types then. Then should we consider to include the type of the pointer? I.e. we should not confuse finger_1 and mouse_1 (on Android, I used a `long` and kept the pointer type in bit 33 :) )

Reply:
> Then should we consider to include the type of the pointer? I.e. we should not confuse finger_1 and mouse_1 

Shouldn't the native pointer id already be different?

Reply:
Depending of the platform not necessarily. As I'm not sure of the behavior for each browser on each OS (yes same browser on different OS sometimes behaves differently), I would try to be the safest possible.

BTW, just realized that I do already have a an helper for that : https://github.com/unoplatform/uno/blob/master/src/Uno.UWP/Devices/Input/PointerIdentifier.cs